powersimdata/scenario/scenario.py

```diff
@@ -41,6 +50,34 @@ def __init__(self, descriptor):
             except AttributeError:
                 return
 
+    def __getattr__(self, name):
+        if name in self.state.exported_methods:
+            return getattr(self.state, name)
+        elif hasattr(self.state, "_custom_getattr_error_message"):
```

Comment:
I'm trying to understand what we are trying achieve here?

Reply:
This is so that if we are in Create state, and we try to get something from the Scenario object that is only available when the builder has been set, we get a more informative message.

Reply:
This is not limited to "Create" state only, right? Do you think the code block I posted will work from this perspective or we have to define this wrapper function `_custom_getattr_error_message` which simply returns `self.__getattr__(name)`?

Reply:
I think the logic is if the scenario is in a `State` other than `Create` and the `name` is a valid attribute/method, it will go through `if` block, otherwise, we are going to raise an AttributeError anyway (a redundant try in this specific situation). Yours is more explicit.

Reply:
Done.